packages/singleton-manager/README.md

```diff
@@ -0,0 +1,297 @@
+- `overlays::overlays::2.x` - version 2.x.x (> 1.0.0 you do 1.x, 2.x)
+- `overlays::overlays::0.10.x` - version 0.10.x (< 1.0.0 you do 0.1.x, 0.2.x)
+
+Examples Don't:
+
+- `overlays` - too generic
+- `overlays::overlays` - you should include a version
+- `overlays::1.x` - you should include a package name & unique var
+- `./index.js::1.x` - it should start with a package name
+
+---
+
+## Singleton Manager Rational
+
+We have an app with 2 pages.
+
+- page-a uses overlays 1.x
+- page-b uses overlays 2.x (gets installed nested)
+
+```txt
+my-app (node_modules)
+├── overlays (1.x)
+├── page-a
+│   └── page-a.js
+└── page-b
+    ├── node_modules
+    │   └── overlays (2.x)
+    └── page-b.js
+```
+
+The tough part in this case is the OverlaysManager within the overlays package as it needs to be a singleton.
+
+It starts of simplified like this
+
+```js
+export class OverlaysManager {
+  name = 'OverlayManager 1.x';
+  blockBody = false;
+  constructor() {
+    this._setupBlocker();
+  }
+  _setupBlocker() {
+    /* ... */
+  }
+  block() {
+    this.blockBody = true; // ...
+  }
+
+  unBlock() {
+    this.blockBody = false; // ...
+  }
+}
+```
+
+## Example A (fail)
+
+See it "fail" e.g. 2 separate OverlaysManager are at work and are "fighting" over the way to block the body.
+
+```bash
+npm run start:fail
+```
+
+Steps to reproduce:
+
+1. Page A click on block
+2. Page B => "Blocked: false" (even when hitting the refresh button)
+
+See [the code](./demo/fail/demo-app.js).
+
+---
+
+## Example B (singleton manager)
+
+The breaking change in `OverlayManager` was renaming of 2 function (which has been deprecated before).
+
+- `block()` => `blockingBody()`
+- `unBlock()` => `unBlockingBody()`
+
+knowing that we can create a Manager that is compatible with both via
+
+```js
+import { OverlaysManager } from 'overlays';
+
+class CompatibleManager extends OverlaysManager {
+  blockingBody() {
+    this.block();
+  }
+  unBlockingBody() {
+    this.unBlock();
+  }
+}
+```
+
+all that is left is a to "override" the default instance of the "users"
+
+```js
+import { singletonManager } from 'singleton-manager';
+
+const manager = new CompatibleManager();
+singletonManager.set('overlays::overlays::1.x', compatibleManager);
+singletonManager.set('overlays::overlays::2.x', compatibleManager);
+```
+
+See it in action
+
+```bash
+npm run start:singleton
+```
+
+and [the code](./demo/singleton/demo-app.js).
+
+---
+
+## Example C (singleton and complex patching on app level)
+
+The breaking change in `OverlayManager` was converting a property to a function and a rename of a function.
+
+- `blockBody` => `_blockBody`
+- `block()` => `blockBody()`
+- `unBlock()` => `unBlockBody()`
+
+e.g. what is impossible to make compatible with a single instance is to have `blockBody` act as a property for 1.x and as a function `blockBody()` for 2.x.
+
+So how do we solve it then?
+
+We will make 2 separate instances of the `OverlayManager`.
```

Comment:
This is really hard to understand. Why are the typeof the same manager -> blockbody different here? Why does making 2 instances help the problem? I think we should see the code inside these docs